### ISSUE TITLE
Type resolution fails to error type for functions re-exported through barrel export * chains

### ISSUE THREAD

**Issue by Lem0naise** [oxc-project/tsgolint#1155]
tsgolint resolves certain function types as `error` when they are imported through a barrel file's `export *` re-exports, causing false positives for `no-unsafe-call`, `no-unsafe-assignment`, `no-unsafe-argument`, and `no-unsafe-return`. `tsc --noEmit` reports zero errors for the same code.

Related to #864 but triggered by barrel re-exports rather than complex generics.

## Reproduction

Monorepo with two packages. The UI package re-exports through a barrel file, consumed by the app.

```
packages/ui/src/lib/toast/use-toast.tsx
  -> export { resetState, useToast, toast };

packages/ui/src/lib/index.ts (barrel, ~70 export * lines)
  -> export * from './toast/use-toast';

apps/web/src/setup.ts
  -> import { resetState } from '@myorg/ui';  // resolves through the barrel
  -> resetState();  // tsgolint: error type, tsc: () => void
```

**Observed:**

```
typescript(no-unsafe-call): Unsafe call of a(n) `error` type typed value.
```

**Expected:** `resetState` should resolve to `() => void`.

A second case: a hook in the app that internally imports from the same barrel also resolves to `error`:

```typescript
// apps/web/src/lib/use-builder.ts
import { useQueryBuilder } from '@myorg/ui'; // via barrel export *
export function useBuilder(opts: BuilderOpts) { /* ... */ }

// apps/web/src/app/tasks/SplitView.tsx
const builder = useBuilder({ initialQuery });
// tsgolint: "Unsafe assignment of an error typed value"
// tsc: correctly inferred return type
```

Both functions are simple - no complex generics involved. The `error` type (rather than `any`) suggests a resolution failure rather than an inference discrepancy.

## Environment

- oxlint: 1.77.0
- oxlint-tsgolint: 7.0.2001
- TypeScript: 7.0.2 (`@typescript/native`)
- pnpm workspaces monorepo
- Barrel file has ~70 `export *` and ~70 named `export { }` re-exports
- `options.typeAware: true` in `.oxlintrc.json`

## Workaround

`oxlint-disable-next-line` inline suppressions on each affected call site.

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue.

Can you:
 - run with `--typecheck`
 - with OXC_LOG=debug
I suspect that we aren't correctly assigning a file to a tsconfig, but without logs, it's hard to exactly say - thanks!

**Comment by Lem0naise** [comment]
Thank you for the quick reply!!! Here are some logs
```
Starting tsgolint
Starting to assign files to programs. Total files: 1
Got tsconfig for file apps/web/src/setupTestFE.ts: apps/web/tsconfig.json
Done assigning files to programs. Total programs: 1. Unmatched files: 0
Program apps/web/tsconfig.json: 1 files
Program created with 6837 source files
```

It looks like the file is assigned to the right `tsconfig`? But then it can't resolve the export:
```
x typescript(TS2305): Module '"discovery-ui"' has no exported member 'resetToasts'.
 ,-[apps/web/src/setupTestFE.ts:7:10]
6 | import { act } from '@testing-library/react';
7 | import { resetToasts } from 'discovery-ui';
   :          ^^^^^^^^^^^
8 | import { afterEach, vi } from 'vitest';
 `----

x typescript(no-unsafe-call): Unsafe call of a(n) `error` type typed value.
 ,-[apps/web/src/setupTestFE.ts:81:5]
80 |   act(() => {
81 |     resetToasts();
    :     ^^^^^^^^^^^
82 |   });
 `----`
```
The export chain is `resetToasts` is exported from the source file, re-exported via `export *` through the barrel and `tsc --noEmit` seems to resolve it with no errors. It may be because the package is `symlinked` through `node_modules`?

(TS 6.0.2 is being used by tac I think. However I think `tsgolint` is using typescript-go 7.0.2)

**Comment by Lem0naise** [comment]
Let me rebuild locally and see - it may be a build config issue or a `tsc` issue and not something that is necessary due to `tsgolint` - it looks like it's excluding this file from my declaration output

**Comment by Lem0naise** [comment]
Embarrassingly an issue on my side not `tsgolint`s! The issue was the macOS was case-insensitive so `tsc` was happy to let me import useScopedQueryBuilder when the file was called usescopedquerybuilder, but `oxc_resolver` was not. Fixing the import path resolved it! 
Thank you nonetheless for your speedy response and if have any more things to add to oxlint I will be sure to contribute 🙏
